op-service/eth/blob.go

```diff
@@ -66,3 +67,67 @@ func KZGToVersionedHash(commitment kzg4844.Commitment) (out common.Hash) {
 func VerifyBlobProof(blob *Blob, commitment kzg4844.Commitment, proof kzg4844.Proof) error {
 	return kzg4844.VerifyBlobProof(*blob.KZGBlob(), commitment, proof)
 }
+
+// FromData encodes the given input data into this blob. The encoding scheme is as follows:
+//
+// First, field elements are encoded as big-endian uint256 in BLS modulus range. To avoid modulus
+// overflow, we can't use the full 32 bytes, so we write data only to the topmost 31 bytes of each.
+// TODO: we can optimize this to get a bit more data from the blobs by using the top byte
```

Comment:
I think the specs say that we do eek out those last 3 bytes per 4 field elements

Reply:
Right, Proto added this to the spec yesterday and it's in the process of implementation in another PR.